symfit/core/fit.py

```diff
@@ -1648,4 +1594,81 @@ def hessian_from_model(model):
         of ``model``.
     """
     jac_model = jacobian_from_model(model, as_functions=True)
-    return jacobian_from_model(jac_model)
\ No newline at end of file
+    return jacobian_from_model(jac_model)
+
+
+def prepare_constraint(relational, extra_params, model_type=CallableModel):
+    """
+    Based on a relational expression, return a Model representing the
+    constraint.
+
+    :param relational: Expression which is a subclass of :class:`~sympy.Rel`.
+    :param extra_params: Parameters for the constraint. Usually this should be
+        the parameters of the corresponding model.
+    :param model_type: Type of model for the constraint.
+    :return: Model representing the constraint.
+    """
+    allowed_types = [sympy.Eq, sympy.Ge, sympy.Le]
+    constraint_type = type(relational)
+    # Check if the type of each constraint is allowed.
+    if constraint_type not in allowed_types:
+        raise ModelError(
+            'Only constraints of the type {} are allowed. A constraint'
+            ' of type {} was provided.'.format(allowed_types,
+                                               constraint_type)
+        )
+    elif constraint_type is sympy.Le:
+        # We change this to a Ge
+        constraint = relational.rhs - relational.lhs
+        constraint_type = sympy.Ge
+    else:
+        constraint = relational.lhs - relational.rhs
+
+    if not issubclass(model_type, BaseNumericalModel):
+        constraint_model = model_type(constraint)
+    else:
```

Comment:
Maybe give all candidates for this a `from_constraint` (or something) method? That would make this bit much cleaner.
See also: factory pattern, https://realpython.com/factory-method-python/

Reply:
You mean an alternative 'constructor'/initializer? That's a good idea.